src/scripts/main.js

```diff
@@ -1,3 +1,69 @@
-'use strict';
+const table = document.querySelector('.field');
+const addRowBtn = document.querySelector('.append-row');
+const removeRowBtn = document.querySelector('.remove-row');
+const addColBtn = document.querySelector('.append-column');
+const removeColBtn = document.querySelector('.remove-column');
 
-// write code here
+const maxLimit = 10;
+const minLimit = 2;
+
+function updateButtonStates() {
+  const rowCount = table.rows.length;
+  const colCount = table.rows[0].cells.length;
+
+  addRowBtn.disabled = rowCount >= maxLimit;
+  removeRowBtn.disabled = rowCount <= minLimit;
+  addColBtn.disabled = colCount >= maxLimit;
+  removeColBtn.disabled = colCount <= minLimit;
+}
+
+function removeRow() {
+  if (table.rows.length > minLimit) {
+    table.deleteRow(-1);
+  }
+  updateButtonStates();
+}
+
+function removeColumn() {
+  if (table.rows[0].cells.length > minLimit) {
+    for (const row of table.rows) {
+      row.deleteCell(-1);
+    }
+  }
+  updateButtonStates();
+}
+
+function addRow() {
+  const rowCount = table.rows.length;
+
+  if (rowCount < maxLimit) {
+    const newRow = table.insertRow();
+
+    for (let i = 0; i < rowCount; i++) {
```

Comment:
The loop should iterate over the number of columns, not the number of rows, to add the correct number of cells to the new row. Change `rowCount` to `colCount` to fix this logic error.